pyproject.toml

```diff
@@ -1,15 +1,30 @@
 [tool.ruff]
-# Never enforce `E501` (line length violations).
-ignore = ["E501", "E741", "W605"]
-select = ["E", "F", "I", "W"]
 line-length = 119
+target-version = "py38"
 
-# Ignore import violations in all `__init__.py` files.
-[tool.ruff.per-file-ignores]
-"__init__.py" = ["E402", "F401", "F403", "F811"]
+[tool.ruff.lint]
+ignore-init-module-imports = true
+extend-select = [
+    "E", # PEP8 errors
+    "F", # PEP8 formatting
+    "I", # Import sorting
+    "W", # PEP8 warnings
+    "UP", # Pyupgrade
+]
+ignore = [
+    "E501", # Line length (handled by ruff-format)
+    "E741", # Ambiguous variable name
+    "W605", # Invalid escape sequence
+    "UP007", # X | Y type annotations
+]
+
+[tool.ruff.lint.per-file-ignores]
+"__init__.py" = [
+    "F401",  # Ignore seemingly unused imports (they're meant for re-export)
```

Comment:
This could be handled by defining `__all__` in the corresponding modules, but I'm fine with simply ignoring those imports.

Reply:
I had a commit for that in here too, but it added +300 lines, so I decided to skip it. Can easily add in a subsequent PR.